File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.py
def solution(storey):
    ans = 0
    
    while storey > 0:
        num = storey % 10
        temp = (storey // 10) % 10
        
        if num > 5 or (num == 5 and temp >= 5):
            ans += (10 - num)
            storey += (10 - num)
        else:
            ans += num
        
        storey //= 10
    
    return ans

File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.java
class Solution {
    public int solution(int storey) {
        int ans = 0;

        while (storey > 0) {
            int num = storey % 10;
            int temp = (storey / 10) % 10;

            if (num > 5 || (num == 5 && temp >= 5)) {
                ans += (10 - num);
                storey += (10 - num);
            } else {
                ans += num;
            }

            storey /= 10;
        }

        return ans;
    }
}